src/decoders/common/src/message_decoder.cpp

```diff
@@ -731,20 +736,28 @@ MessageDecoderBase::Decode(const unsigned char* pucMessage_, std::vector<FieldCo
             return STATUS::NO_DEFINITION;
         }
 
-        pvCurrentMsgFields = vMsgDef->GetMsgDefFromCrc(pclMyLogger, stMetaData_.uiMessageCrc);
+        pvCurrentMsgFields = vMsgDef->GetMsgDefFromCrc(*pclMyLogger, stMetaData_.uiMessageCrc);
     }
 
     // Expand the intermediate format vector to prevent the copy constructor from being called when the vector grows in size
     stInterMessage_.clear();
-    stInterMessage_.reserve(pvCurrentMsgFields->size());
+    stInterMessage_.reserve(pvCurrentMsgFields.size());
 
     // Decode the detected format
-    return stMetaData_.eFormat == HEADER_FORMAT::ASCII || stMetaData_.eFormat == HEADER_FORMAT::SHORT_ASCII
-               ? DecodeAscii<false>(*pvCurrentMsgFields, reinterpret_cast<const char**>(&pucTempInData), stInterMessage_)
-           : stMetaData_.eFormat == HEADER_FORMAT::ABB_ASCII || stMetaData_.eFormat == HEADER_FORMAT::SHORT_ABB_ASCII
-               ? DecodeAscii<true>(*pvCurrentMsgFields, reinterpret_cast<const char**>(&pucTempInData), stInterMessage_)
-           : stMetaData_.eFormat == HEADER_FORMAT::BINARY || stMetaData_.eFormat == HEADER_FORMAT::SHORT_BINARY
-               ? DecodeBinary(*pvCurrentMsgFields, &pucTempInData, stInterMessage_, stMetaData_.uiBinaryMsgLength)
-           : stMetaData_.eFormat == HEADER_FORMAT::JSON ? DecodeJson(*pvCurrentMsgFields, json::parse(pucTempInData)["body"], stInterMessage_)
-                                                        : STATUS::UNKNOWN;
+    switch (stMetaData_.eFormat)
+    {
+    case HEADER_FORMAT::ASCII: [[fallthrough]];
+    case HEADER_FORMAT::SHORT_ASCII: //
+        return DecodeAscii<false>(pvCurrentMsgFields, reinterpret_cast<const char**>(&pucTempInData), stInterMessage_);
+    case HEADER_FORMAT::ABB_ASCII: [[fallthrough]];
+    case HEADER_FORMAT::SHORT_ABB_ASCII: //
+        return DecodeAscii<true>(pvCurrentMsgFields, reinterpret_cast<const char**>(&pucTempInData), stInterMessage_);
+    case HEADER_FORMAT::BINARY: [[fallthrough]];
+    case HEADER_FORMAT::SHORT_BINARY: //
+        return DecodeBinary(pvCurrentMsgFields, &pucTempInData, stInterMessage_, stMetaData_.uiBinaryMsgLength);
+    case HEADER_FORMAT::JSON: //
+        return DecodeJson(pvCurrentMsgFields, json::parse(pucTempInData)["body"], stInterMessage_);
+    default: //
+        return STATUS::UNKNOWN;
```

Comment:
```suggestion
    case HEADER_FORMAT::ASCII: [[fallthrough]];
    case HEADER_FORMAT::SHORT_ASCII: return DecodeAscii<false>(pvCurrentMsgFields, reinterpret_cast<const char**>(&pucTempInData), stInterMessage_);
    case HEADER_FORMAT::ABB_ASCII: [[fallthrough]];
    case HEADER_FORMAT::SHORT_ABB_ASCII: return DecodeAscii<true>(pvCurrentMsgFields, reinterpret_cast<const char**>(&pucTempInData), stInterMessage_);
    case HEADER_FORMAT::BINARY: [[fallthrough]];
    case HEADER_FORMAT::SHORT_BINARY: return DecodeBinary(pvCurrentMsgFields, &pucTempInData, stInterMessage_, stMetaData_.uiBinaryMsgLength);
    case HEADER_FORMAT::JSON: return DecodeJson(pvCurrentMsgFields, json::parse(pucTempInData)["body"], stInterMessage_);
    default: return STATUS::UNKNOWN;
```

Reply:
I'd keep it as is. The empty comments are there to ensure the autoformatter formats the non-fallthrough cases consistently for readability.